src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -2702,9 +2713,176 @@ namespace SimAirServingZones {
+		// TRUE when air loop has been evaluated with latest actuated variables
+		bool IsUpToDateFlag;
+		// Iteration counter
+		static int Iter( 0 );
+		// Number of times that the maximum iterations was exceeded
+		static int ErrCount( 0 );
+		// Number of times that the maximum iterations was exceeded
+		static int MaxErrCount( 0 );
+		// Placeholder for environment name used in error reporting
+		static std::string ErrEnvironmentName;
+		// A character string equivalent of ErrCount
+		std::string CharErrOut;
+		int static AirLoopPass;
+		static bool BypassOAController;
+
+		// FLOW:
+
+		BypassOAController = false; // simulate OA water coil controllers
+		AirLoopPass = AirLoopControlInfo( AirLoopNum ).AirLoopPass;
+		IsUpToDateFlag = false;
+		PrimaryAirSystem( AirLoopNum ).ControlConverged = false;
+
+		AllowWarmRestartFlag = true;
+		AirLoopControlInfo( AirLoopNum ).AllowWarmRestartFlag = true;
+
+		// This call to ManageControllers reinitializes the controllers actuated variables to zero
+
+		// BypassOAController is false here since we want to simulate the controller
+		ManageControllers( ControllerName, ControllerIndex, FirstHVACIteration, AirLoopNum, iControllerOpColdStart, ControllerConvergedFlag, IsUpToDateFlag, BypassOAController, AllowWarmRestartFlag );
+
+		// Detect whether the speculative warm restart feature is supported by each controller on this air loop.
+		AirLoopControlInfo( AirLoopNum ).AllowWarmRestartFlag = AirLoopControlInfo( AirLoopNum ).AllowWarmRestartFlag && AllowWarmRestartFlag;
+
+		// Evaluate water coils with new actuated variables
+		if ( HXAssistedWaterCoil ) {
+			SimHXAssistedCoolingCoil( CompName, FirstHVACIteration, CoilOn, 0.0, CompIndex, ContFanCycCoil );
+		} else {
+			SimulateWaterCoilComponents( CompName, FirstHVACIteration, CompIndex );
+		}
+		IsUpToDateFlag = true;
+
+		// Loop over the air sys controllers until convergence or MaxIter iterations
+		Iter = 0;
+		ControllerConvergedFlag = false;
+		// if the controller can be locked out by the economizer operation and the economizer is active, leave the controller inactive
+		if ( AirLoopControlInfo( AirLoopNum ).EconoActive && PrimaryAirSystem( AirLoopNum ).CanBeLockedOutByEcono( ControllerIndex ) ) {
+			ControllerConvergedFlag = true;
+		}
+
+		// For this controller, iterate until convergence
+		while ( !ControllerConvergedFlag ) {
+
+			++Iter;
+
+			ManageControllers( ControllerName, ControllerIndex, FirstHVACIteration, AirLoopNum, iControllerOpIterate, ControllerConvergedFlag, IsUpToDateFlag, BypassOAController );
+
+			PrimaryAirSystem( AirLoopNum ).ControlConverged( ControllerIndex ) = ControllerConvergedFlag;
+
+			if ( !ControllerConvergedFlag ) {
+				// Only check abnormal termination if not yet converged
+				// The iteration counter has been exceeded.
+				if ( Iter > MaxIter ) {
+
+					// The warning message will be suppressed during the warm up days.
+					if ( !WarmupFlag ) {
+						++ErrCount;
+						if ( ErrCount < 15 ) {
```

Comment:
15. Again I'm just being curious if this was a trial and error value or if it just made sense to use this number.

Reply:
This new function is very similar to SolveAirLoopControllers, just focuses on a single air loop instead of looping through all of them (maybe could have done something trick with the existing function, I didn't go back once I got it working). 15 is also used there so I didn't change it. These warnings could be spread out over the year instead of successive starting at some point in time so we print X of them before handing over to ShowRecurringWarningErrorAtEnd.